src/misc/translate-from-firestore.ts

```diff
@@ -121,19 +121,29 @@ export const recursivelyMapStorageUrls = async (
       (fieldValue as any[]).map(async (value, index) => {
         fieldValue[index] = await recursivelyMapStorageUrls(fireWrapper, value);
       })
-    );
+    ).then(() => {
+      return fieldValue;
+    });
   }
   const isDocumentReference = isInputADocReference(fieldValue);
   if (isDocumentReference) {
     return fieldValue;
   }
+
+  const isJSDate = fieldValue instanceof Date;
+  if(isJSDate){
+    return fieldValue;
+  }
```

Comment:
What version of `firebase` are you using? I didn't think the firestore API would ever return a `Date` field 🤔 

It is only meant to return a `Timestamp` which should be handled by this part here:

https://github.com/benwinding/react-admin-firebase/blob/5a6bdec6d069bdc9b2c87a51adf3ca96766d36d8/src/misc/translate-from-firestore.ts#L47-L50